### PR TITLE
Prepare for 3.1

### PR DESCRIPTION
**Pull request by levitte**
Because we now have an openssl-3.0 branch, master is moved to be the
next potential minor version.

### CODE REVIEW

**Comment by levitte** [comment]
I prepared this with the release script, as follows:

``` console
$ ./dev/release.sh --branch --no-upload --no-update
```

But then simply threw away everything related to the release branch and only kept the master branch commit.

File: NEWS.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ OpenSSL Releases
 OpenSSL 3.0
```
Comment (romen) [inline NEWS.md:18]: Uhm, it seems this didn't work that well.

No 3.1 link above in the list, no new 3.1 section, and the new subsection is inside the 3.0 section

File: CHANGES.md [hunk 2]
```diff
@@ -28,6 +28,10 @@ breaking changes, and mappings for the large list of deprecated functions.
 
 [Migration guide]: https://github.com/openssl/openssl/tree/master/doc/man7/migration_guide.pod
 
+### Changes between 3.0 and 3.1 [xx XXX xxxx]
```
Comment (romen) [inline CHANGES.md:31]: Similar to the other comment: https://github.com/openssl/openssl/pull/16484/files#r699715510

File: NEWS.md (hunk 1 above)
Comment (levitte) [reply NEWS.md:18]: Argh, I haven't payed attention to the sectioning introduced in 4477beacc4104335642fd99bd421592a9c07af21...

**Comment by levitte** [comment]
Good catch, @romen.  I will fix them manually for this, then I'll have to fix the fixup scripts...

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Merged

2727265752c66690d79c4cbe6956746977b1df4c Prepare for 3.1

**Pull request closed without merging**